Source/adventureDAO.js

```diff
@@ -383,29 +377,37 @@ exports.endRound = async function (adventure, thread) {
 			})
 
 			// Replace placeholder
-			adventure.room.moves.splice(adventure.room.moves.findIndex(move => move.userReference.team === "enemy" && move.userReference.index == index), 1);
-			if (counterpartHasPriority) {
-				adventure.room.priorityMoves.push(move);
-			} else {
-				adventure.room.moves.push(move);
+			let placeholderIdx=adventure.room.moves.findIndex(move => move.userReference.team === "enemy" && move.userReference.index == index)
+			if(placeholderIdx >=0){
+				adventure.room.moves.splice(placeholderIdx, 1,move);
 			}
 		}
 	});
 
 
 	// Randomize speed ties
-	[adventure.room.priorityMoves, adventure.room.moves].forEach(moveQueue => {
-		moveQueue.forEach(move => {
-			move.speed += generateRandomNumber(adventure, 10, "battle") / 10;
-		})
-		moveQueue.sort((first, second) => {
-			return second.speed - first.speed;
-		})
+	let randomOrderBag=Array(adventure.room.moves.length).fill().map((_,idx)=>idx) // ensure that unique values are available for each move
```

Comment:
```suggestion
	let randomOrderBag=Array(adventure.room.moves.length).map((_,idx)=>idx) // ensure that unique values are available for each move
```
Didn't verify, but I don't think the `fill()` is necessary?

Reply:
toying around with node console a bit it seems like fill is necessary (otherwise it just returns 10 empty items.... which I do find kind of odd)

Reply:
Yeah, huh, I thought it would map the indices of the empty items. Does that not work?

Reply:
<img width="210" alt="image" src="https://github.com/Imaginary-Horizons-Productions/prophets-of-the-labyrinth/assets/7379741/43922768-8e80-446a-ada3-f6f590974923">
🤷

Reply:
I agree that I'd've thought empty items to be iterable

Reply:
Huh. "Make an array of X length" is just so Python-esque that I never kept up with the intricacies. Good to know, I guess.